frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt

```diff
@@ -0,0 +1,74 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.layout.buttonGroup
+import com.intellij.ui.layout.panel
+import com.virtuslab.gitmachete.backend.api.IGitMacheteBranch
+import com.virtuslab.gitmachete.backend.api.IGitMacheteNonRootBranch
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString
+import java.text.MessageFormat.format
+
+enum class OverrideOption {
+  PARENT,
+  INFERRED
+}
+
+class OverrideForkPointDialog
+    constructor(
+        project: Project,
+        private val parentBranch: IGitMacheteBranch,
+        private val branch: IGitMacheteNonRootBranch
+    ) : DialogWrapper(project, /* canBeParent */ true) {
+
+  private var myOverrideOption = OverrideOption.PARENT
+
+  init {
+    title =
+        getString("action.GitMachete.BaseOverrideForkPointAction.dialog.override-fork-point.title")
+    setOKButtonMnemonic('O'.toInt())
+    init()
+  }
+
+  fun showAndGetSelectedCommit() =
+      if (!showAndGet()) null
+      else
+          when (myOverrideOption) {
+        OverrideOption.PARENT -> parentBranch.pointedCommit
+        OverrideOption.INFERRED -> branch.forkPoint.orNull
+      }
+
+  override fun createCenterPanel() =
+      panel {
+        row(
+            getString(
+                "action.GitMachete.BaseOverrideForkPointAction.dialog.override-fork-point.label.branch-name")) {
+          label(branch.name, bold = true)
+        }
+        buttonGroup(::myOverrideOption) {
+          row {
+            radioButton(
+                format(
+                    getString(
+                        "action.GitMachete.BaseOverrideForkPointAction.dialog.override-fork-point.radio-button.parent"),
+                    parentBranch.name),
+                OverrideOption.PARENT,
+                /* comment */ (parentBranch.pointedCommit.shortMessage
+                    ?: "cannot resolve commit message"))
```

Comment:
```
w: /home/pawellipski/git-machete-intellij-plugin/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt: (57, 21): Elvis operator (?:) always returns the left operand of non-nullable type String
```

Reply:
Btw pls add an analogue of `-Werror` flag to Kotlin setup to promote all warnings to errors